src/App.tsx

```
@@ -1,27 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum BtnType {
+  Nothing = '',
+  All = 'All',
+  Five = 'Five',
+  Red = 'Red',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [changeListOfGoods, setChangeListOfGoods] = useState<BtnType>();
+  const [listGoods, setListGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  useEffect(() => {
+    let items: Promise<Good[]> | undefined;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    switch (changeListOfGoods) {
+      case BtnType.All:
+        items = getAll();
+        break;
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      case BtnType.Five:
+        items = get5First();
+        break;
+
+      case BtnType.Red:
+        items = getRedGoods();
+        break;
+
+      default:
+        items = undefined;
+        break;
+    }
+
+    items?.then(item => setListGoods(item)).catch();
```

This is a good start for handling errors, and it prevents the app from crashing. However, an empty `catch` block silently swallows the error, which can make debugging difficult. To fully meet the requirement of handling errors (Checklist Item #3), it's best practice to at least log the error to the console, like so: `.catch(console.error)`.